src/UniTask/Assets/Plugins/UniTask/Runtime/UniTaskExtensions.cs

```diff
@@ -28,7 +28,7 @@ public static UniTask<T> AsUniTask<T>(this Task<T> task, bool useCurrentSynchron
                         p.TrySetCanceled();
                         break;
                     case TaskStatus.Faulted:
-                        p.TrySetException(x.Exception);
+                        p.TrySetException(x.Exception.InnerException);
```

Comment:
Even if there are multiple InnerExceptions, always unwrapping.
This is  `await Task.WhenAll(...).AsUniTask()`, the behavior is the same as `await Task.WhenAll()`.
Is this a problem?

Reply:
However, it is more natural to get multiple exceptions when WhenAll.
I ended up going for the same behavior as `await Task.WhenAll`, but maybe I should break it and still provide AggregateException.
Sorry, but I would like to discuss this a bit.